egs/wsj/s5/utils/scoring/wer_per_spk_details.pl

```diff
@@ -130,6 +130,7 @@ sub format_sys {
 }
 
 open(UTT2SPK,$ARGV[0]) or die "Could not open the utt2spk file $ARGV[0]";
+binmode(UTT2SPK, ":utf8");
```

Comment:
Thanks for finding the problem.
However, I don't think we can assume that the utt2spk is utf-8, because it might be in other encodings as well.  I think this should use the get_utf8_or_bytestream function.  However it would require code changes in that loop, and better if someone who knows perl does this.  Actually this is reaching the limits of my perl knowledge; I don't know what backslash-asterisk means.  @jtrmal is there any chance you could work on this, or tell us what the correct fix is?

Reply:
backslash-asterisk is just a way how to generate reference for the special file handles that are not variables (for example OUTPUT from open(OUTPUT, "r") but not $OUTPUT from open($OUTPUT, "r")), if that's making things clearer

I think the correct fix might be something like (I didn't test)
```
(my $utt_is_utf8, my @utt_lines) = get_utf8_or_bytestream(\*UTT2SPK);
die "Cannot read file" unless defined(@utt_lines);
while (@utt_lines) {
  my $line = shift @utt_lines;
  my @F=split $line;
  die "Incompatible format of the utt2spk file: $_" if @F != 2;
  $UTTMAP{$F[0]} = $F[1];
  # Set width of speaker column by its longest label,
  if($SPK_WIDTH < length($F[1])) { $SPK_WIDTH = length($F[1]) 
}
close(UTT2SPK);
```

Reply:
I've updated the commit, used proposed, made some cosmetic changes.
now the output is like this (it works for me)

# cat exp/tri3/decode_sp/scoring_kaldi/penalty_1.0/17.txt | align-text --special-symbol='***' ark:exp/tri3/decode_sp/scoring_kaldi/test_filt.txt ark:- ark,t:- | utils/scoring/wer_per_utt_details.pl --special-symbol '***' | tee exp/tri3/decode_sp/scoring_kaldi/wer_details/per_utt | utils/scoring/wer_per_spk_details.pl data/test/utt2spk > exp/tri3/decode_sp/scoring_kaldi/wer_details/per_spk
# Started at Fri Nov  2 11:46:44 +03 2018
#
align-text '--special-symbol=***' ark:exp/tri3/decode_sp/scoring_kaldi/test_filt.txt ark:- ark,t:-
LOG (align-text[5.5.88~1417-8e30f]:main():align-text.cc:129) Done 829 sentences, failed for 0
utils/scoring/wer_per_spk_details.pl: Note: handling as utf-8 text
utils/scoring/wer_per_utt_details.pl: Note: handling as utf-8 text
utils/scoring/wer_per_spk_details.pl: Note: handling as utf-8 text
# Accounting: time=0 threads=1
# Ended (code 0) at Fri Nov  2 11:46:44 +03 2018, elapsed time 0 seconds